apps/freeablo/faworld/monster.h

```diff
@@ -10,18 +11,30 @@ namespace DiabloExe
 
 namespace FAWorld
 {
+    class AI;
+    class NullAI;
+
     class Monster: public Actor
     {
         STATIC_HANDLE_NET_OBJECT_IN_CLASS()
 
         public:
             Monster();
             Monster(const DiabloExe::Monster& monster, Position pos);
+            ~Monster();
             void init();
             std::string getDieWav();
             std::string getHitWav();
+            void attachAI(AI* ai) {
+              mAi = ai;
+            };
+
+            bool attack(Actor * enemy);
+            void update(bool noclip, size_t ticksPassed);
+            bool canIAttack(Actor * actor);
 
         private:
+            AI * mAi;
```

Comment:
I've added the `NullAI` for this reason mostly, it simply does nothing so actor can initialize to that and monsters can then initialize some other behaviour.

I'll move it up, it makes sense since other entities will also have their AIs (like towns people or cows :D though I don't remember if they wandered around).